MediaBrowser.Model/Entities/ProviderIdsExtensions.cs

```diff
@@ -137,7 +155,25 @@ public static void SetProviderId(this IHasProviderIds instance, string name, str
         /// <param name="value">The value.</param>
         public static void SetProviderId(this IHasProviderIds instance, MetadataProvider provider, string value)
         {
-            instance.SetProviderId(provider.ToString(), value);
+            if (instance == null)
+            {
+                throw new ArgumentNullException(nameof(instance));
+            }
+
+            var providerString = provider.ToString();
+
+            // If it's null remove the key from the dictionary
+            if (string.IsNullOrEmpty(value))
+            {
+                instance.ProviderIds?.Remove(providerString);
+            }
+            else
+            {
+                // Ensure it exists
+                instance.ProviderIds ??= new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                instance.ProviderIds[providerString] = value;
+            }
```

Comment:
Why duplicate the logic?
`instance.SetProviderId(str, value)` already does this

Reply:
Because that function is used a lot more often all over the code than the one with the string argument.
It skips the Enum matching.

Reply:
Why should it skip?

Reply:
Because the input is already a value of that enum. Runnign toString on it and passing it to the function with the string arguent would just match that string against the wole enum again which is unnecessary.

Reply:
It's a dictionary lookup with an `O(1)` time complexity. The code duplication is much worse.